api/envoy/extensions/filters/http/compressor/v3/compressor.proto

```diff
@@ -21,40 +21,94 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // Compressor :ref:`configuration overview <config_http_filters_compressor>`.
 // [#extension: envoy.filters.http.compressor]
 
-// [#next-free-field: 7]
+// [#next-free-field: 9]
 message Compressor {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.compressor.v2.Compressor";
 
+  message CommonDirectionConfig {
+    // Runtime flag that controls whether the filter is enabled for compression or not. If set to false, the
+    // filter will operate as a pass-through filter. If the field is omitted, the filter will be enabled.
+    config.core.v3.RuntimeFeatureFlag enabled = 1;
+
+    // Minimum response length, in bytes, which will trigger compression. The default value is 30.
+    google.protobuf.UInt32Value content_length = 2;
+
+    // Set of strings that allows specifying which mime-types yield compression; e.g.,
+    // application/json, text/html, etc. When this field is not defined, compression will be applied
+    // to the following mime-types: "application/javascript", "application/json",
+    // "application/xhtml+xml", "image/svg+xml", "text/css", "text/html", "text/plain", "text/xml"
+    // and their synonyms.
+    repeated string content_type = 3;
+  }
+
+  // Configuration for filter behavior on the request direction.
+  message RequestDirectionConfig {
+    CommonDirectionConfig common_config = 1;
+  }
+
+  // Configuration for filter behavior on the response direction.
+  message ResponseDirectionConfig {
+    CommonDirectionConfig common_config = 1;
+
+    // If true, disables compression when the response contains an etag header. When it is false, the
+    // filter will preserve weak etags and remove the ones that require strong validation.
+    bool disable_on_etag_header = 2;
+
+    // If true, removes accept-encoding from the request headers before dispatching it to the upstream
```

Comment:
If this is request header then it should be part of `RequestDirectionConfig`?

Reply:
Not sure about that, because it makes sense only in the context of compressing response payloads, that is in the "response direction".

Reply:
oh I see, nvm.